web/src/components/layout/layout.tsx

```diff
@@ -75,22 +75,23 @@ class Layout extends React.PureComponent<LayoutProps, LayoutState> {
   private installApp: HTMLElement;
 
   state: LayoutState = {
-    challengeTeamToken: challengeTeamTokens.find(challengeTeamToken =>
-      this.props.location.search.includes(`team=${challengeTeamToken}`)
-    ),
-    challengeToken: challengeTokens.find(challengeToken =>
-      this.props.location.search.includes(`challenge=${challengeToken}`)
-    ),
+    challengeTeamToken: undefined,
+    challengeToken: undefined,
     isMenuVisible: false,
     hasScrolled: false,
     hasScrolledDown: false,
     showStagingBanner: isStaging(),
-    showWelcomeModal: true,
+    showWelcomeModal: false,
   };
 
   componentDidMount() {
     this.scroller.addEventListener('scroll', this.handleScroll);
     this.visitHash();
+    this.setState({
+      challengeTeamToken: this.getTeamToken(),
+      challengeToken: this.getChallengeToken(),
```

Comment:
Nitpick, but I think calling each of these methods twice is unnecessary.

```javascript
componentDidMount() {
  ...
  const challengeToken = this.getChallengeToken();
  const challengeTeamToken = this.getTeamToken();
  this.setState({
    challengeToken,
    challengeTeamToken,
    showWelcomeModal: challengeToken && challengeTeamToken,
  });
}
```

Reply:
Much cleaner, thanks!